src/main/java/lotto/domain/LottoNumber.java

```diff
@@ -2,23 +2,38 @@
 
 import lotto.exception.InvalidLottoNumberException;
 
+import java.util.HashMap;
+import java.util.Map;
 import java.util.Objects;
 
 public class LottoNumber implements Comparable<LottoNumber> {
+    private static final Map<Integer, LottoNumber> LOTTO_NUMBER_CACHE = new HashMap<>();
+
+    static {
+        for (int i = LottoFactory.MIN_LOTTO_NUMBER, len = LottoFactory.MAX_LOTTO_NUMBER; i <= len; i++) {
+            LOTTO_NUMBER_CACHE.put(i, new LottoNumber(i));
+        }
+    }
+
     private Integer lottoNumber;
 
-    LottoNumber(Integer lottoNumber) {
-        checkRangeOfLottoNumber(lottoNumber);
+    private LottoNumber(Integer lottoNumber) {
         this.lottoNumber = lottoNumber;
     }
 
-    void checkRangeOfLottoNumber(int lottoNumber) {
+    static void checkRangeOfLottoNumber(int lottoNumber) {
         if (checkNumberRange(lottoNumber, LottoFactory.MIN_LOTTO_NUMBER, LottoFactory.MAX_LOTTO_NUMBER)) {
             throw new InvalidLottoNumberException(lottoNumber);
         }
     }
 
-    private boolean checkNumberRange(int lottoNumber, int minNumber, int maxNumber) {
+    public static LottoNumber valueOf(int lottoNumber) {
+        checkRangeOfLottoNumber(lottoNumber);
+        new LottoNumber(lottoNumber);
+        return LOTTO_NUMBER_CACHE.get(lottoNumber);
```

Comment:
범위 체크를 매번 해줄 필요가 있을지 의문이네요.

아래와 같이 표현 할 수도 있어요.
```suggestion
        return Optional.ofNullable(LOTTO_NUMBER_CACHE.get(lottoNumber))
                .orElseThrow(() -> new InvalidLottoNumberException(lottoNumber));
```

Reply:
예제 코드 감사드립니다! 직접 보니 어떤 식으로 사용하는지 알 것 같습니다. 사실, 자바 8을 잘 모르긴 하지만 2단계 과제까지는 자바 8 이전 코드로 연습해보고자 했습니다.🙂